stubs/docker/docker/models/images.pyi

```diff
@@ -1,45 +1,46 @@
-from _typeshed import Incomplete
-from collections.abc import Generator
-from typing import Any
+from collections.abc import Iterator, Sequence
+from typing import Any, Literal, overload
 
 from .resource import Collection, Model
 
 class Image(Model):
     @property
-    def labels(self): ...
+    def labels(self) -> dict[str, Any]: ...
     @property
-    def short_id(self): ...
+    def short_id(self) -> str: ...
     @property
-    def tags(self): ...
-    def history(self): ...
-    def remove(self, force: bool = False, noprune: bool = False): ...
-    def save(self, chunk_size=2097152, named: bool = False): ...
-    def tag(self, repository, tag: str | None = None, **kwargs): ...
+    def tags(self) -> Sequence[str]: ...
+    def history(self) -> Sequence[Any]: ...
+    def remove(self, force: bool = False, noprune: bool = False) -> dict[str, Any]: ...
+    def save(self, chunk_size: int = 2097152, named: bool = False) -> Iterator[Any]: ...
+    def tag(self, repository: str, tag: str | None = None, **kwargs) -> bool: ...
 
 class RegistryData(Model):
-    image_name: Incomplete
-    def __init__(self, image_name, *args, **kwargs) -> None: ...
+    image_name: str
+    def __init__(self, image_name: str, *args, **kwargs) -> None: ...
     @property
-    def id(self): ...
+    def id(self) -> str: ...
     @property
-    def short_id(self): ...
-    def pull(self, platform: Incomplete | None = None): ...
+    def short_id(self) -> str: ...
+    def pull(self, platform: str | None = None) -> Image: ...
     def has_platform(self, platform): ...
-    attrs: Incomplete
     def reload(self) -> None: ...
 
-class ImageCollection(Collection):
+class ImageCollection(Collection[Image]):
     model: type[Image]
-    def build(self, **kwargs) -> tuple[Image, Generator[Any, None, None]]: ...
-    def get(self, name): ...
-    def get_registry_data(self, name, auth_config: Incomplete | None = None): ...
-    def list(self, name: Incomplete | None = None, all: bool = False, filters: Incomplete | None = None): ...
-    def load(self, data): ...
-    def pull(self, repository, tag: str | None = None, all_tags: bool = False, **kwargs): ...
-    def push(self, repository, tag: str | None = None, **kwargs): ...
+    def build(self, **kwargs) -> tuple[Image, Iterator[Any]]: ...
+    def get(self, name: str) -> Image: ...
+    def get_registry_data(self, name, auth_config: dict[str, Any] | None = None) -> RegistryData: ...
+    def list(self, name: str | None = None, all: bool = False, filters: dict[str, Any] | None = None) -> Sequence[Image]: ...
+    def load(self, data: bytes) -> Sequence[Image]: ...
+    @overload
+    def pull(self, repository: str, tag: str | None = None, all_tags: Literal[False] = False, **kwargs) -> Image: ...
+    @overload
+    def pull(self, repository: str, tag: str | None = None, *, all_tags: Literal[True], **kwargs) -> Sequence[Image]: ...
```

Comment:
This requires a third overload so that `all_tags` can be supplied positionally:

```python
@overload
def pull(self, repository: str, tag: str | None, all_tags: Literal[True], **kwargs) -> list[Image]: ...
```

Also return a concrete list.